runtime/parser2/expression.go

```diff
@@ -626,6 +615,95 @@ func defineLessThanOrTypeArgumentsExpression() {
 		})
 }
 
+func defineGreaterThanOrBitwiseRightShiftExpression() {
+
+	setExprMetaLeftDenotation(
+		lexer.TokenGreater,
+		func(p *parser, rightBindingPower int, left ast.Expression) (result ast.Expression, done bool) {
+
+			// If the right binding power is higher than any of the potential cases,
+			// then return early
+
+			if rightBindingPower >= exprLeftBindingPowerBitwiseShift &&
+				rightBindingPower >= exprLeftBindingPowerComparison {
+
+				return left, true
+			}
+
+			// Start buffering before skipping the `>` token,
+			// so it can be replayed in case the right binding power
+			// was higher than the determined left binding power.
```

Comment:
```suggestion
			// Since `>` and `>>` share a common token, we need to distinguish them by buffering a `>`
			// and see if the next is another `>`:
			// if yes, it's a `>>`; if no, then it's a `>`.
			//
			// buffering allows us to replay the buffered tokens in case the right
			// binding power was higher than the determined left binding power.
```